### PR TITLE
ldap: reduce the amount of #ifdefs needed

### PR DESCRIPTION
**Pull request by bagder**

### CODE REVIEW

File: lib/ldap.c [hunk 1]
```diff
@@ -259,6 +259,13 @@ static int ldap_win_bind(struct connectdata *conn, LDAP *server,
 }
 #endif /* #if defined(USE_WIN32_LDAP) */
 
+#if defined(USE_WIN32_LDAP)
+#define FREE_ON_WINLDAP(x) curlx_unicodefree(x)
```
Comment (emilengler) [inline lib/ldap.c:263]: Shouldn't the curlx_unicodefree((c)) into two brackets like it is in other places

Comment (bagder) [reply lib/ldap.c:263]: ("brackets" are usually these: `[]` - I don't think you meant those)

I don't understand. Why do we need more parentheses?

Comment (emilengler) [reply lib/ldap.c:263]: I mean these () brackets. They are being used in other macros in the code as well. See Curl_safefree()

**Comment by dfandrich** [comment]
The brackets elsewhere have a functional effect. They're there to prevent
issues with operator precedence in case the macro is called with an expression
instead of a bare identifier. In this context they would accomplish nothing.

**Pull request closed without merging**